src/network/networkpacket.cpp

```
@@ -81,6 +81,24 @@ void NetworkPacket::putRawString(const char* src, u32 len)
 	m_read_offset += len;
 }
 
+void NetworkPacket::putCString(const char* src, u32 len)
+{
+	u16 msgsize = len;
+	if (msgsize > 0xFFFF) {
+		msgsize = 0xFFFF;
```

Silently truncating data is bad...